### DISCUSSION TITLE
How to display the file like image and video into frontend after picking with the OpenFileDialog?

### DISCUSSION THREAD

**Discussion by dunkbing** [wailsapp/wails#4198] [Q&A]
I want to pick a file (image, video), perform some operations on it, and then display it to the front-end. If I do the picking from the front-end, then I cannot get the file path.

**Comment by yalay** [comment]
use runtime.OpenMultipleFilesDialog to pick a file

**Comment by yalay** [comment]
refer to https://wails.io/docs/reference/runtime/dialog

**Comment by dunkbing** [comment]
> use runtime.OpenMultipleFilesDialog to pick a file

This function only returns the selected files 😅 How do I display it on the screen?

**Comment by yalay** [comment]
You mean  runtime.BrowserOpenURL(f.ctx, "file://"+filePath)?

**Comment by leaanthony** [comment]
Either return the whole file via the binding (probably slow) or what I'd do is setup an [asset handler](https://wails.io/docs/reference/options#handler) to return the file and then after selection of the file emit an event to load it, which will trigger the assethandler to load it.

So:
1. Install asset handler to manage `/fs/` paths
2. Create JS event handler which listens to "fileload" events and does a fetch on "/fs/"+filename
3. When you select a file, emit ("fileload", filename)

It's not ideal and I'd be cautious of opening your whole filesystem up so maybe configure your asset handler to process paths only in a certain sub directory
